### PR TITLE
[Ng Jun Sheng] iP

### PR DESCRIPTION
**Pull request by jsincorporated**
# DukeTasks

> "aikennut dewit" - Prof Daron Tan [(source)](https://www.comp.nus.edu.sg/~cs1101x/img/photos/tantc.gif)

DukeTasks frees your mind of having to remember things you need to do. It's,

- CLI based
- easy to use
- ~~FAST~~ good speed👍

All you need to do is:

1. Download it from [here](https://github.com/jsincorporated/ip/releases).
2. Double click it
3. Add your tasks
4. Let is remember them for you 😎

And it is **free**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines(coming soon)
- [ ] Managing reminders(coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

### CODE REVIEW

File: src/main/java/duke/Event.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + "(at: " + at + ")";
+    }
+
+    @Override
+    public String toUser() {
```
Comment (vantemoon) [inline src/main/java/duke/Event.java:23]: Perhaps a more intuitive name for this method? And for other overwritten/overwriting methods as well.

File: src/main/java/duke/EventHandler.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+     *
+     * @param input User input
+     * @throws DukeNoDescriptionException Thrown when description is empty.
+     */
+    public String addDeadline(String input) throws DukeNoDescriptionException {
+        if (input.length() == 8) {
+            throw new DukeNoDescriptionException();
+        }
+
+        int endAt = input.indexOf("/");
```
Comment (vantemoon) [inline src/main/java/duke/EventHandler.java]: Maybe endIndex can be a better choice here?

File: src/main/java/duke/Task.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package duke;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Comment (vantemoon) [inline src/main/java/duke/Task.java:5]: I like how you named this boolean value to make it sounds like a boolean. 👍

File: src/main/java/duke/TaskList.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    public void remove(int index) {
+        taskList.remove(index);
+    }
+
+    public ArrayList<Task> getList() {
+        return this.taskList;
+    }
+
+    public TaskList filterByKeyword(String query) {
```
Comment (vantemoon) [inline src/main/java/duke/TaskList.java:34]: I like how you used the name to explain what the method does.

**Review by vantemoon — COMMENTED** [review]
Looks good to me:) Overall good code quality.

File: src/main/java/duke/Task.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    public void markAsDone() { isDone = true; }
+
+    public void markAsUndone() { isDone = false; }
+
+    public static Task fromString(String data) {
+        if (data.charAt(1) == 'T') {
+            return ToDo.fromString(data);
+        }
+        switch (data.charAt(1)) {
+            case 'T':
```
Comment (yilinzyl) [inline src/main/java/duke/Task.java]: Maybe you could try to align the case and switch statements!

File: src/main/java/duke/Task.java (hunk 3 above)
Comment (yilinzyl) [reply src/main/java/duke/Task.java:5]: Agreed!

File: src/main/java/duke/Storage.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.File;
+import java.util.Scanner;
```
Comment (yilinzyl) [inline src/main/java/duke/Storage.java:7]: Maybe you consider adding a line spacing between different import groups?

**Review by yilinzyl — COMMENTED** [review]
Overall code is clean and follows coding standard! LGTM :)

File: src/main/java/duke/Duke.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    ;
+
+    public static void main(String[] args) throws DukeException {
+//        new Duke("data/tasks.txt").run();
+        new Duke("data/tasks.txt");
+    }
+
+//    public void run() throws DukeException {
```
Comment (AugustDespair) [inline src/main/java/duke/Duke.java]: I think that you should delete this block of code if you do not intend on using it in future increments.

File: src/main/java/duke/Duke.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        }
+    }
+
+    public Duke() {
+    }
+
+    ;
+
+    public static void main(String[] args) throws DukeException {
+//        new Duke("data/tasks.txt").run();
```
Comment (AugustDespair) [inline src/main/java/duke/Duke.java]: IIRC, style guide wants comments to be intended according to the line beneath it, so you might wanna move the `//` to be next to `new Duke`.

File: src/main/java/duke/EventHandler.java (hunk 2 above)
Comment (AugustDespair) [reply src/main/java/duke/EventHandler.java]: I agree with this. endIndex sounds more descriptive imo.

File: src/main/java/duke/Task.java (hunk 5 above)
Comment (AugustDespair) [reply src/main/java/duke/Task.java]: I agree, I think the style guide wants us to do that as well.

**Review by AugustDespair — COMMENTED** [review]
Your code is really neat. Nice!

**Pull request closed without merging**